Anastasia_PCS/Homework #1.js

```diff
@@ -0,0 +1,252 @@
+//Homework #1
+
+//(Due Monday 7/20)
+
+//You need to turn in this homework by cloning this repo, making a new branch with your solutions, 
+//pushing to github, and issuing a pull request. For details, follow the instructions here.//
+
+//Please also read the learning objectives for this week.
+
+//1) (30% of total time) Predict the output of each of the following expressions, and explain 
+//the rules which determine that answer. (Hint: The rules for operator == are complicated, 
+//but do your best!)
+
+//If the output depends on the value or type of variable x, identify the conditions 
+//(what x is) when the expression will output true (or false, if that's simpler). Assume the cases 
+//are independent, and x is reset to an unknown value before each.
+
+//Some of these are tricky! Don't trust your first instinct.//
+
+a) "1" == 1	//TRUE - compares value only
+
+b) "1" === 1 //FALSE - compares value & type
+
+c) x == 'x' //TRUE - compares value only
+
+d) x == (x+'') //TRUE - compares value only
+
+e) '' == ' ' //FALSE - an empty string is "falsey" - Also, "undefined" == "string"
+
+f) x = true //TRUE 
+
+g) var x; x == 'undefined' //FALSE - x is not == to 'undefined'
+
+h) '9'<'10' //FALSE - a string cannot be < or > another string
+
```

Comment:
Strings can indeed be greater or less than one another. The characters are compared in order and then evaluated. In this case javascript finds '9' greater than '1' so it stops its comparison there and then returns false (because '9' > '1'). Likewise '942' > '9260' because '9' == '9', but '4' > '2', and then javascript stops comparison, having come to a greater value on either side, which returns true.